decoder/src/lib.rs

```diff
@@ -1025,18 +1025,38 @@ fn format_args_real(
                     Arg::IStr(x) => format_str(x, hint, &mut buf)?,
                     Arg::Format { format, args } => buf.push_str(&format_args(format, args, hint)),
                     Arg::FormatSlice { elements } => {
-                        buf.write_str("[")?;
-                        let mut is_first = true;
-                        for element in elements {
-                            if !is_first {
-                                buf.write_str(", ")?;
+                        match hint {
+                            Some(DisplayHint::Ascii) => {
+                                let vals = elements
+                                    .iter()
+                                    .filter_map(|e| match e.args[0] {
+                                        Arg::Uxx(v) => Some(u8::try_from(v).ok()?),
+                                        _ => None,
+                                    })
+                                    .collect::<Vec<u8>>();
+                                format_bytes(&vals, hint, &mut buf)?
```

Comment:
When formatting a slice of structs (or anything that isn't a `u8`) with `{=[?]:a}`, I'd expect the display hint to be ignored, but the code looks like it'll instead output nothing.

Reply:
For example, you might want to format this struct with `{=[?]:a}` in order to display the `name` field as an ASCII string:

```rust
struct Data<'a> {
    name: &'a [u8],
    value: bool,
}
```